oak/server/wasm_node.cc

```diff
@@ -39,6 +39,11 @@ namespace oak {
 static std::unique_ptr<wabt::FileStream> s_log_stream = wabt::FileStream::CreateStdout();
 static std::unique_ptr<wabt::FileStream> s_stdout_stream = wabt::FileStream::CreateStdout();
 
+static bool MemoryAvailable(wabt::interp::Environment* env, const uint32_t offset,
```

Comment:
Thanks, this makes sense, but it still seems easy to forget or use incorrect offset / size here. Do you think it would be possible eventually to have functions to read memory that return StatusOr or some other Option-like type?

Reply:
Something `Option`-like would definitely reduce the chance of mistakes, but it would have the downside that error paths would end up being scattered through the code (e.g. `channel_read` needs to check 4 different memory areas).  Might well be worth it, though.